modules.d/95udev-rules/module-setup.sh

```diff
@@ -55,8 +54,16 @@ install() {
     inst_rules 91-permissions.rules
     # eudev rules
     inst_rules 80-drivers-modprobe.rules
-    # legacy persistent network device name rules
-    [[ $hostonly ]] && inst_rules 70-persistent-net.rules
+
+    # only include persistent network device name rules if network is set up
+    # in the initrd
+    # Avoid interference with systemd predictable network device naming
+    if dracut_module_included "network" || dracut_module_included "network-manager"; then
```

Comment:
Is it necessary to check if `network-manager` is included?

Reply:
In my test with 055, dracut_module_included "network"  only covered network-legacy, so I've added network-manager.
But yes, maybe that should better be fixed in the network module definition.
IIRC, in 059, it covers all network modules.

Reply:
In 055 the `network` meta module includes a specific network module depending on the system installation (if a specific network module is not added via `--add`):

https://github.com/openSUSE/dracut/blob/dcb9bdbf70650051957e2b93863ab14aab87c47e/modules.d/40network/module-setup.sh#L19-L29

E.g., I'm using NetworkManager on my Leap 15.4 (dracut-055), so:

```
# dracut -f --stdlog 3 -a network test.img
# lsinitrd -m test.img | grep net
network-manager
network
kernel-network-modules
```

Reply:
Hm ok, for me  **dracut_module_included network**  didn't work when I added the network-manager module via config file:

`# cat /etc/dracut.conf.d/59-network.conf 
add_dracutmodules+=" network-manager "
`

The test doesn't recognized it as network module:

`/usr/lib/dracut/modules.d/95udev-rules/module-setup.sh@61(install): dracut_module_included network
/usr/lib/dracut/dracut-init.sh@161(dracut_module_included): [[   udev-rules  fs-lib  dracut-systemd  haveged  usrmount  base  shutdown  suse  suse-initrd   bash systemd systemd-initrd dbus-daemon dbus i18n network-manager drm plymouth btrfs kernel-modules kernel-modules-extra qemu resume rootfs-block suse-btrfs suse-xfs terminfo   == *\ network\ * ]]`

Reply:
The `network` module adds other network modules, but not the other way around. The specific network modules (`systemd-networkd`, `network-manager`, `network-legacy`) do not add the `network` meta module. E.g. adding `network-legacy`:

```
# dracut -f --stdlog 3 -a network-legacy test.img
# lsinitrd -m test.img | grep net
network-legacy
```

Reply:
Ah ok, so, I will use the specific names for the test.
Thanks, updated the commit.